### ISSUE TITLE
value assigned to `found` is never read, in loop with continue

### ISSUE THREAD

**Issue by Thomblin** [rust-lang/rust-clippy#13023]
### Summary

For the given code I get the warning "value assigned to `found` is never read", however the code is working fine. I guess clippy get'S confused with two nested loops combined with the continue statement.

### Lint Name

unused_assignments

### Reproducer

I tried this code:

```rust
fn main() {
    'cells: for i in 0..3 {
        let mut found = false;
        for _ in [1, 2] {
            if i == 2
            {
                println!("found");
                found = true;
                continue 'cells;
            }
        }
        if !found {
            println!("not found");
        }
    }
}
```

I saw this happen:

```
   Compiling playground v0.0.1 (/playground)
warning: value assigned to `found` is never read
 --> src/main.rs:8:17
  |
8 |                 found = true;
  |                 ^^^^^
  |
  = help: maybe it is overwritten before being read?
  = note: `#[warn(unused_assignments)]` on by default

warning: `playground` (bin "playground") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.73s
     Running `target/debug/playground`
```

I expected to see this happen:

clippy should not emit a warning

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: x86_64-unknown-linux-gnu
release: 1.78.0
LLVM version: 18.1.2
```

**Comment by Alexendoo** [comment]
`unused_assignments` is from the rust compiler rather than clippy

It's correct here though, the assigned `true` is never read because it's followed by a continue to the outer loop, `found` is always `false` if it hits the line `if !found {`

**Comment by Thomblin** [comment]
ah well. yes I guess I had a false reasoning here
